tests/e2e/tests/simple/testdata/nginx/nginx-injected.yaml.tmpl

```diff
@@ -0,0 +1,369 @@
+          readOnlyRootFilesystem: true
+          runAsUser: 1337
+        volumeMounts:
+        - mountPath: /etc/istio/proxy
+          name: istio-envoy
+        - mountPath: /etc/certs/
+          name: istio-certs
+          readOnly: true
+      initContainers:
+      - args:
+        - -p
+        - "15001"
+        - -u
+        - "1337"
+        image: gcr.io/nathanmittler-istio-test/proxy_init:latest
+        imagePullPolicy: Always
+        name: istio-init
+        resources: {}
+        securityContext:
+          capabilities:
+            add:
+            - NET_ADMIN
+      - args:
+        - -c
+        - sysctl -w kernel.core_pattern=/etc/istio/proxy/core.%e.%p.%t && ulimit -c
+          unlimited
+        command:
+        - /bin/sh
+        image: alpine
+        imagePullPolicy: Always
+        name: enable-core-dump
+        resources: {}
+        securityContext:
+          privileged: true
+      terminationGracePeriodSeconds: 60
+      volumes:
+      - emptyDir:
+          medium: Memory
+        name: istio-envoy
+      - name: istio-certs
+        secret:
+          optional: true
+          secretName: istio.default
+status: {}
+---
+
+# Service: nginx-default-http-backend
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx-default-http-backend
+  labels:
+    app: nginx-default-http-backend
+spec:
+  ports:
+  - name: http
+    port: 80
+    targetPort: http
+  selector:
+    app: nginx-default-http-backend
+---
+
+# Deployment: nginx-ingress-controller
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  creationTimestamp: null
+  name: nginx-ingress-controller
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: ingress-nginx
+  strategy: {}
+  template:
+    metadata:
+      annotations:
+        prometheus.io/port: "10254"
+        prometheus.io/scrape: "true"
+        sidecar.istio.io/inject: "false"
+        sidecar.istio.io/status: '{"version":"edd82d936380ef27ac55181113282e5ef29e7e36c205c194b566569389ebb74b","initContainers":["istio-init","enable-core-dump"],"containers":["istio-proxy"],"volumes":["istio-envoy","istio-certs"]}'
+      creationTimestamp: null
+      labels:
+        app: ingress-nginx
+    spec:
+      containers:
+      - args:
+        - /nginx-ingress-controller
+        - --default-backend-service=$(POD_NAMESPACE)/nginx-default-http-backend
+        - --configmap=$(POD_NAMESPACE)/nginx-configuration
+        - --tcp-services-configmap=$(POD_NAMESPACE)/nginx-tcp-services
+        - --udp-services-configmap=$(POD_NAMESPACE)/nginx-udp-services
+        - --annotations-prefix=nginx.ingress.kubernetes.io
+        - --v=10
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        image: quay.io/kubernetes-ingress-controller/nginx-ingress-controller:0.12.0
```

Comment:
Q: each cloud provider typically has its own nginx ingress controller image... Could you clarify how you are proposing other cloud provide to leverage this envoy side car feature like the yaml shows here?

Reply:
Sorry what I meant to ask is how would user get the envoy side car injected for nginx?     Is it same as `istioctl kube-inject` today being used for microservices?  I can't tell from the examples as it is already being injected by the yaml.

Reply:
Yes, I generated this file originally by using `istioctl kube-inject`. The next step after this PR goes in will be to automate some of this a bit more, so manual injection isn't necessary.